enclave/core/CMakeLists.txt

```diff
@@ -1,79 +1,106 @@
+        sgx/memory.c
+        sgx/once.c
+        sgx/properties.c
+        sgx/report.c
+        sgx/sbrk.c
+        sgx/spinlock.c
+        sgx/td.c
+        sgx/thread.c
+        sgx/enter.S
+        sgx/exit.S
+        sgx/getkey.S
     )
-elseif (WIN32)
-set(PLATFORM_SRC
-    ../../common/rand.asm
-    windows/reloc.c
+
+    # OS specific sources for SGX.
+    if (UNIX)
+        list(APPEND PLATFORM_SRC
+            ../../common/sgx/rand.S
+            sgx/linux/reloc.c
+            sgx/linux/threadlocal.c
+        )
+    elseif (WIN32)
+        list(APPEND PLATFORM_SRC
+            ../../common/sgx/rand.asm
+            sgx/windows/reloc.c
+        )
+    endif()
+elseif(OE_TRUSTZONE)
+    message("TODO: ADD ARM files.")
+    list(APPEND PLATFORM_SRC
+        optee/abort.c
+        optee/bounds.c
+        optee/host.c
+        optee/printf.c
+        optee/start.c
     )
 endif()
 
 add_library(oecore STATIC
     ../../common/safecrt.c
     assert.c
-    atexit.c
-    backtrace.c
-    calls.c
-    tracee.c
-    cpuid.c
-    debugmalloc.c
-    entropy.c
-    exception.c
-    globals.c
-    hostcalls.c
     hexdump.c
-    init.c
-    jump.c
-    keys.c
-    malloc.c
-    memory.c
-    once.c
-    properties.c
     result.c
-    report.c
-    sbrk.c
     snprintf.c
-    spinlock.c
     string.c
-    td.c
-    thread.c
     time.c
-    enter.S
-    exit.S
-    getkey.S
     ${PLATFORM_SRC}
-    )
+)
 
 target_link_libraries(oecore PUBLIC oe_includes)
 
-# This directory contains dummy C headers needed by dlmalloc:
-target_include_directories(oecore PRIVATE ${CMAKE_CURRENT_SOURCE_DIR}/dlmalloc)
-
 if (CMAKE_C_COMPILER_ID MATCHES GNU)
     target_compile_options(oecore PRIVATE -Wjump-misses-init)
-endif ()
+endif()
+
+if (OE_SGX)
+    # This directory contains dummy C headers needed by dlmalloc:
+    target_include_directories(oecore PRIVATE ${CMAKE_CURRENT_SOURCE_DIR}/sgx/dlmalloc)
+
+    # Unfortunately dlmalloc uses GNU extension that allows arithmetic
+    # null pointers.
+    set_source_files_properties(sgx/malloc.c
+        PROPERTIES COMPILE_FLAGS "-Wno-conversion -Wno-null-pointer-arithmetic")
 
-# Unfortunately dlmalloc uses GNU extension that allows arithmetic
-# null pointers.
-set_source_files_properties(
-    malloc.c 
-    PROPERTIES COMPILE_FLAGS "-Wno-conversion -Wno-null-pointer-arithmetic")
+    # jump.s must be optimized for the correct call-frame.
+    set_source_files_properties(sgx/jump.c PROPERTIES COMPILE_FLAGS -O2)
 
-set_source_files_properties(keys.c PROPERTIES COMPILE_FLAGS -Wno-type-limits)
+    set_source_files_properties(sgx/keys.c PROPERTIES COMPILE_FLAGS -Wno-type-limits)
+
+    # -m64 is an x86_64 specific flag
+   target_compile_options(oecore PUBLIC -m64)
+endif()
+
+if (CMAKE_C_COMPILER_ID MATCHES GNU)
+    target_compile_options(oecore PRIVATE -Wjump-misses-init)
+endif ()
 
 target_compile_options(oecore PUBLIC
-    -m64
     -fPIC
     -nostdinc
     -fno-stack-protector
     -fvisibility=hidden
-    )
+)
```

Comment:
If you're updating style, I tend to prefer the closing paren to be on the same line as the last item. It's silly all on its own line 😛

Reply:
sure